A110-child-channel-plugins.md

```diff
@@ -0,0 +1,215 @@
+# **A110: Child Channel Configuration Plugins**
+
+* **Author(s)**: [Abhishek Agrawal](mailto:agrawalabhi@google.com)
+* **Status**: Draft
+* **To be implemented in**: Core, Java, Go
+* **Last updated**: 2025-12-23
+
+## **Abstract**
+
+This proposal introduces a mechanism to configure "child channels"—channels created internally by gRPC components (such as xDS control plane channel). Currently, these internal channels cannot easily inherit configuration (like metric sinks and interceptors) from the user application without relying on global state. This design proposes a language-specific approach to injecting configuration: using `functional interfaces` in Java, `DialOptions` in Go, and `ChannelArgs` in Core.
+
+## **Background**
+
+Complex gRPC ecosystems often require the creation of auxiliary channels that are not directly instantiated by the user application. Two primary examples are:
+
+1. **xDS (Extensible Discovery Service)**: When a user creates a channel with an xDS target, the gRPC library internally creates a separate channel to communicate with the xDS control plane. Currently, users have limited ability to configure this internal control plane channel.
```

Comment:
In the xDS case, the same xDS channel will be used in multiple channels for the same target.  We need to document what the behavior will be if the application creates multiple channels to the same target with different child channel args.  (It will just use whatever child channel args are passed to the first channel to be created.)